pkg/util/net/http.go

```diff
@@ -67,6 +70,9 @@ func SetTransportDefaults(t *http.Transport) *http.Transport {
 	if t.TLSHandshakeTimeout == 0 {
 		t.TLSHandshakeTimeout = defaultTransport.TLSHandshakeTimeout
 	}
+	if err := http2.ConfigureTransport(t); err != nil {
```

Comment:
In almost every client config we create in cluster (anything in a pod) the
config will have a custom CA, which will trigger a custom TLSClientConfig,
which would prevent http2 from being enabled by default.

On Mon, May 9, 2016 at 1:23 PM, krousey <notifications@github.com> wrote:

> In pkg/util/net/http.go
> <https://github.com/kubernetes/kubernetes/pull/25280#discussion_r62537120>
> :
>
> > @@ -67,6 +70,9 @@ func SetTransportDefaults(t *http.Transport) *http.Transport {
> >  	if t.TLSHandshakeTimeout == 0 {
> >  		t.TLSHandshakeTimeout = defaultTransport.TLSHandshakeTimeout
> >  	}
> > +	if err := http2.ConfigureTransport(t); err != nil {
>
> I don't understand how a custom CA affects or is affected by http2. What
> am I missing?
>
> —
> You are receiving this because you are on a team that was mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/25280/files/53ddc93c6e80e750d09fa72680c718754e0693bc#r62537120>
>